docs/localization.md

```diff
@@ -0,0 +1,116 @@
+# Writing Localizeable Code
+
+As part of Microsoft's commitment to global-ready products, our developer tools are localized into 14 different languages.
+For this effort, vcpkg has hooks in order to allow our localization team to translate all messages that are shown to the user by the vcpkg tool.
+As a developer on vcpkg, the only thing you have to consider is the messages in the C++ source files,
+and `locales/messages.json` -- everything else will be generated and modified by the localization team.
+
+## Declaring a Message
+
+The process of writing a message to be displayed to the user starts with the message; if it needs to be in a header,
+use the `DECLARE_MESSAGE(<message-name>, <parameters>, <comment>, <english-message>` in that header,
+followed by `REGISTER_MESSAGE(<message-name>)` in a source file; if it can just be in a source file,
+use `DECLARE_AND_REGISTER_MESSAGE()`.
```

Comment:
I think you should work in a complete example to provide context for the subheadings. For example, a direct link to `https://github.com/microsoft/vcpkg-tool/blob/13a09ef0359e259627d46560a22a6e182730da7b/src/vcpkg/sourceparagraph.cpp#L24-L28`

which looks like this when rendered by GitHub:

https://github.com/microsoft/vcpkg-tool/blob/13a09ef0359e259627d46560a22a6e182730da7b/src/vcpkg/sourceparagraph.cpp#L24-L28